### DISCUSSION TITLE
ESP32-c3 invalid header: 0xffffffff

### DISCUSSION THREAD

**Discussion by PeterNoelEvans** [micropython/micropython#9359] [Hardware & Peripherals]
I have an issue with my "ESP32-C3-DevKitM-1 V1.0" (printed on the back).

Originally it was working fine and I would code with micropython from Thonny without issue. One day I started playing around with ESP-IDF and uploaded the example code for "Hello World" and it would print "Hello World" but kept rebooting every 10 seconds with a count down. I didn't like that so tried to go back to Thonny and Micropython again and now I keep getting this error and it won't boot.

```
rst:0x7 (TG0WDT_SYS_RST),boot:0xc (SPI_FAST_FLASH_BOOT)
Saved PC:0x4004d1f8
invalid header: 0xffffffff
```

I tried flashing it again with esptool and everything seems to work:

```
C:\Users\Admin\Downloads>esptool.py --chip esp32c3 -p COM5 --before=default_reset --after=hard_reset write_flash --flash_mode dio --flash_freq 80m --flash_size 4MB 0x0 c:\Users\Admin\Downloads\esp32c3-20220618-v1.19.1.bin
esptool.py v3.2
Serial port COM5
Connecting....
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 34:b4:72:01:21:00
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x0015efff...
Compressed 1437472 bytes to 872137...
Wrote 1437472 bytes (872137 compressed) at 0x00000000 in 77.7 seconds (effective 148.0 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

I checked it 
```
C:\Users\Admin\Downloads>esptool.py verify_flash --diff yes 0x0 c:\Users\Admin\Downloads\esp32c3-20220618-v1.19.1.bin
esptool.py v3.2
Found 2 serial ports
Serial port COM5
Connecting....
Detecting chip type... ESP32-C3
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 34:b4:72:01:21:00
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Verifying 0x15ef20 (1437472) bytes @ 0x00000000 in flash against c:\Users\Admin\Downloads\esp32c3-20220618-v1.19.1.bin...
-- verify OK (digest matched)
Hard resetting via RTS pin...
```

but it still wouldn't boot for me.

**Comment by Lobo-T** [comment]
Try using esptool to erase the whole flash before flashing micropython again.

**Reply by PeterNoelEvans** [reply]
Done that several times with esptool and Thonny.
I have actually done everything several times including trying different bin files and I tried ESP-IDF on several different occasions which produced various different kinds of errors but nothing that would indicate that it was a solution.

**Reply by puppet13th** [reply]
currently i used esptool version 3.3 to flash ai-thinker esp-c3-32s and lilygo t-io plus without issue. if there is boot button,try to manually enter boot mode before flashing.
have u successfully flash mpy using esptool v3.2 before?

**Reply by puppet13th** [reply]
btw i suspect the spi flash has died.
to check:
1. erase flash
2. dump flash with
`esptool -p comXXX read_flash 0x0 0x400000 spi_flash.bin` --> this should produce 4MB file.

sha256sum should be
`cd3517473707d59c3d915b52a3e16213cadce80d9ffb2b4371958fb7acb51a08`

**Reply by PeterNoelEvans** [reply]
Thank you kindly for your help.


```
>>> ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x1 (POWERON),boot:0xc (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd6100,len:0xe3c
load:0x403ce000,len:0x6f4
load:0x403d0000,len:0x28d8
entry 0x403ce000
MicroPython v1.19.1 on 2022-06-18; ESP32C3 module with ESP32C3
Type "help()" for more information.
```

I finally got it working.

**Reply by puppet13th** [reply]
I'm curious, what have u done to brought it back to the living land?

**Reply by PeterNoelEvans** [reply]
Yes, I am curious too. Frankly, I don't really know.
I erased the flash again with erase flash then...
I pasted in ```esptool -p comXXX read_flash 0x0 0x400000 spi_flash.bin``` although I didn't know where the spi_flash.bin was going to come from but I did it anyway.

It did some stuff that took quite some time. It through back a few errors.

I then flashed the esp32c3.bin file that I downloaded and bingo. I opened Thonny and there it was.

**Reply by puppet13th** [reply]
that command is used to dump esp32 spi flash content to file.tinkeringtime then.

**Reply by PeterNoelEvans** [reply]
I think I am going to give on this unit now. It has gone back to the 0xffffffff error again.
I am pretty sure the hardware is faulty rather than anything else. I have half a dozen other ESP32 boards that give me no major issues and so it is really only this one. I only have one ESP32-C3.

**Comment by puppet13th** [comment]
esp32-c3 flash start at 0x0

`esptool -p comXXX write_flash 0x0 firmware.bin`

works for me

edit: i did not notice 0x0 part.

erase before flash new firmware with
`esptool -p comXXX erase_flash`

**Reply by PeterNoelEvans** [reply]
Thank you. Yes, I erased the flash with esptool just like this several times too.

**Reply by PeterNoelEvans** [reply]
Now I have it working again, I feel that it was a nice exercise in testing patience and learning some new things about esptool.
